packages/client/__tests__/pages/ResourceIndex.spec.js

```diff
@@ -59,13 +71,103 @@ describe('Test the resource index page', () => {
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            fireEvent.click(screen.getAllByTestId('table-row')[0])
+        )
+    })
+    test('can search for values on the resource table', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            expect(screen.queryAllByTestId('table-row')).toHaveLength(2)
+        )
+        fireEvent.change(screen.getByTestId('search-resource'), {
+            target: { value: 'Amapai' },
+        })
+        window.Flamingo = {
+            ...FlamingoMock,
+            request: {
+                get: jest.fn().mockResolvedValue({
+                    data: {
+                        data: [
+                            {
+                                av_cpc: 21,
+                                category: 'angular',
+                                content: 'this is the content for amapai',
+                                created_at: '2020-08-12T21:40:24.000Z',
+                                description:
+                                    'this is the description for amapai',
+                                id: 1,
+                                published_at: '2020-08-11T23:00:00.000Z',
+                                scheduled_for: '2020-08-11T23:00:00.000Z',
+                                title: 'Amapai',
+                                updated_at: '2020-08-12T21:40:24.000Z',
+                                user_id: 2,
+                            },
+                        ],
+                        page: 1,
+                        total: 1,
+                        perPage: 10,
+                        pageCount: 1,
+                    },
+                }),
+            },
+        }
+        await waitFor(() =>
+            expect(screen.queryAllByTestId('table-row')).toHaveLength(1)
+        )
+    })
+    test('table row should show correct amount of data', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            expect(screen.queryAllByTestId('table-row')).toHaveLength(2)
+        )
+    })
+    test('clicking on the delete icon should delete that specific row data', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            fireEvent.click(screen.getAllByTestId('delete-icon')[0])
+        )
+    })
+    test('clicking on the filter button should open the filter dropdown', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        expect(screen.queryByTestId('filter-box')).toBeFalsy()
```

Comment:
Please space your assertions where necessary so that it's as clear as possible. 

For example, 

```
fireEvent.click(screen.queryByTestId('filter-button'))
expect(screen.queryByTestId('filter-box')).toBeTruthy()
```
These two should be grouped, then after that space because they are related.